src/TropicalGeometry/hypersurface.jl

```diff
@@ -0,0 +1,260 @@
+###
+# Tropical hypersurfaces in Oscar
+# ===============================
+###
+
+
+
+###
+# 0. Definition
+# -------------
+# M = typeof(min) or typeof(max):
+#   min or max convention, affecting initial forms, tropical polynomial, etc.
+# EMB = true or false:
+#   embedded or abstract tropical hypersurface
+#   embedded tropical variety = weighted polyhedral complex in euclidean space
+#   abstract tropical variety = weighted hypergraph with enumerated vertices
+###
+
+@attributes mutable struct TropicalHypersurface{M,EMB} <: TropicalVarietySupertype{M,EMB}
+    polyhedralComplex::PolyhedralComplex
+    function TropicalHypersurface{M,EMB}(Sigma::PolyhedralComplex) where {M,EMB}
+        if codim(Sigma)!=1
+            error("TropicalHypersurface: input polyhedral complex not one-codimensional")
+        end
+        return new{M,EMB}(Sigma)
+    end
+end
+export TropicalHypersurface
+
+function pm_object(T::TropicalHypersurface)
+    if has_attribute(T,:polymake_bigobject)
+        return get_attribute(T,:polymake_bigobject)
+    end
+    error("pm_object(T::TropicalHypersurface): Has no polymake bigobject")
+end
+
+
+
+###
+# 1. Printing
+# -----------
+###
+
+function Base.show(io::IO, th::TropicalHypersurface{M, EMB}) where {M, EMB}
+    if EMB
+        print(io, "A $(repr(M)) tropical hypersurface embedded in $(ambient_dim(th))-dimensional Euclidian space")
+    else
+        print(io, "An abstract $(repr(M)) tropical hypersurface of dimension $(dim(th))")
+    end
+end
+
+
+
+###
+# 2. Basic constructors
+# ---------------------
+###
+
+@doc Markdown.doc"""
+    TropicalHypersurface(f::MPolyElem{TropicalSemiringElem})
+
+Return the tropical hypersurface of a tropical polynomial.
+
+# Examples
+```jldoctest
+julia> T = tropical_semiring(min)
+Tropical ring (min)
+
+julia> Txy,(x,y) = T["x","y"]
+(Multivariate Polynomial Ring in x, y over Tropical ring (min), AbstractAlgebra.Generic.MPoly{Oscar.TropicalSemiringElem{typeof(min)}}[x, y])
+
+julia> f = x+y+1
+x + y + (1)
+
+julia> Tf = TropicalHypersurface(f)
+A min tropical hypersurface embedded in 2-dimensional Euclidian space
+```
+"""
+function TropicalHypersurface(
+    f::Union{AbstractAlgebra.Generic.MPoly{Oscar.TropicalSemiringElem{typeof(min)}},
+             AbstractAlgebra.Generic.MPoly{Oscar.TropicalSemiringElem{typeof(max)}}}
+        )
```

Comment:
How about this (more compact, easier to read)
```suggestion
    f::AbstractAlgebra.Generic.MPoly{Oscar.TropicalSemiringElem{T}}) where T
```